relational_operators/AggregationOperator.hpp

```diff
@@ -103,23 +109,23 @@ class AggregationWorkOrder : public WorkOrder {
    * @brief Constructor
    *
    * @param input_block_id The block id.
-   * @param aggr_state_index The index of the AggregationState in QueryContext.
+   * @param state The AggregationState to use.
    **/
   AggregationWorkOrder(
       const block_id input_block_id,
-      const QueryContext::aggregation_state_id aggr_state_index)
+      AggregationOperationState *state)
```

Comment:
Thanks for the clarification. Let me know if the following interpretation is correct - 

The WorkOrders will be constructed on the master node with all the information that they need (e.g. StorageManager, any operator state etc.). 
If the WorkOrders need to be scheduled on a remote node, the WorkOrderFactory will serialize them and replace the pointers with some IDs.
On the remote node, the WorkOrders will be deserialized using the WorkOrderFactory methods. This will involve replacing the IDs with actual pointers for that node.